newIDE/app/src/ExportAndShare/BrowserExporters/BrowserS3PreviewLauncher/index.js

```diff
@@ -219,10 +230,58 @@ export default class BrowserS3PreviewLauncher extends React.Component<
       );
 
       // If the preview windows are new, register them so that they can be accessed
-      // by the debugger.
+      // by the debugger and for the captures to be detected when they close.
       if (!existingPreviewWindow) {
         previewWindows.forEach((previewWindow: WindowProxy) => {
-          registerNewPreviewWindow(previewWindow);
+          const debuggerId = registerNewPreviewWindow(previewWindow);
+          browserPreviewDebuggerServer.registerCallbacks({
+            onErrorReceived: () => {},
+            onServerStateChanged: () => {},
+            onConnectionClosed: async ({ id }) => {
+              if (id !== debuggerId) {
+                return;
+              }
+
+              if (previewOptions.captureOptions) {
+                const { screenshots } = previewOptions.captureOptions;
+                if (!screenshots) return;
+                const screenshotPublicUrls: string[] = screenshots.map(
+                  screenshot => screenshot.publicUrl
+                );
+
+                // Check if they have been properly uploaded.
+                const responseUploadedScreenshotPublicUrls: Array<
+                  string | null
+                > = await Promise.all(
+                  screenshotPublicUrls.map(
+                    async (screenshotUrl): Promise<string | null> => {
+                      const response = await fetch(screenshotUrl, {
```

Comment:
Can this throw? If so, it should be caught because we want to catch and consider this as a failure but not bubble up to the app.